### PR TITLE
add additional privacy and legal information re: Semgrep Assistant

### PR DESCRIPTION
**Pull request by khorne3**
adds [privacy and legal info](https://docs.google.com/document/d/1e3hXwhnxs6MOtGabqJ6FDasTxgGGI8LLXG894U3DBHM/edit#heading=h.9b3t6qpuci67) re: Assistant

### Please ensure

- [x] A subject matter expert (SME) reviews the content

### CODE REVIEW

File: docs/semgrep-code/semgrep-assistant-code.md
Comment (s-santillan) [inline docs/semgrep-code/semgrep-assistant-code.md:60]:
```suggestion
* Semgrep, Inc. is a paying customer of OpenAI and has a Data Protection Agreement signed with them (provided upon request by [contacting support](/docs/support). The code snippets we upload are be persisted by OpenAI temporarily, following their data usage policies at [Open AI API data Usage Policies](https://openai.com/policies/api-data-usage-policies).
```

File: docs/semgrep-code/semgrep-assistant-code.md
Comment (s-santillan) [inline docs/semgrep-code/semgrep-assistant-code.md:61]:
```suggestion
* Semgrep, Inc. takes several steps to protect data that is processed by AI because Assistant requires the sharing of code snippets with a third party. Firstly, Semgrep Assistant shares code snippets with OpenAI without identifying the customer or repository name. Secondly, Semgrep only shares the code necessary to enlist the help of GPT in automating the resolution of each specific alert. Finally, Semgrep Assistant only accesses source code repositories on a file-by-file basis; it does not need or request org-level access to your codebase.
```

This should probably be a list. Long paragraphs are difficult for users to skim.

File: docs/semgrep-code/semgrep-assistant-code.md
Comment (s-santillan) [inline docs/semgrep-code/semgrep-assistant-code.md:62]:
```suggestion
* When using Semgrep Assistant, source code **does** leave your repository; Semgrep submits part of the file with a finding to OpenAI for processing by a GPT model. OpenAI is not allowed to use the submitted code for training their models.
```

File: docs/semgrep-code/semgrep-assistant-code.md
Comment (s-santillan) [inline docs/semgrep-code/semgrep-assistant-code.md:65]:
```suggestion
* Because Semgrep Assistant accesses OpenAI’s services through the API, OpenAI does not use any of the code provided to them to improve their services (see Section 3(c) of their Terms of Use).
```

Google Style Guide forbids "via".

File: docs/semgrep-code/semgrep-assistant-code.md
Comment (s-santillan) [inline docs/semgrep-code/semgrep-assistant-code.md:64]:
```suggestion
* Semgrep, Inc. and OpenAI do not obtain any rights to your source code. Your source code remains yours, and Semgrep or OpenAI accesses it to the limited extent necessary to provide the Semgrep Assistant service. Once the results are returned to you, Semgrep Assistant deletes the shared snippets. OpenAI retains copies of the content sent to them for a maximum of 30 days to monitor for abuse, as indicated in their API Data Usage Policies.
```

File: docs/semgrep-code/semgrep-assistant-code.md
Comment (s-santillan) [inline docs/semgrep-code/semgrep-assistant-code.md:66]:
```suggestion
* To a limited extent, using Semgrep Assistant changes the terms of your agreement with Semgrep, Inc. Specifically, sharing code snippets with Semgrep Assistant as part of this feature expands the scope of the data to which you grant Semgrep, Inc. a limited license to provide services to you (see Section 5.1 of our Subscriber Agreement).
```

**Comment by s-santillan** [comment]
This is a supporting article to a legal document, and in a legal document **actors** must be defined. For this reason, I have had to specify which "Semgrep" is performing the action - whether it is Semgrep Inc or Semgrep Assistant.

File: docs/semgrep-code/semgrep-assistant-code.md
Comment (s-santillan) [inline docs/semgrep-code/semgrep-assistant-code.md:61]:
```suggestion
* Semgrep, Inc. takes the following steps to protect data that is processed by AI since Assistant requires the sharing of code snippets with a third party:
```

File: docs/semgrep-code/semgrep-assistant-code.md
Comment (s-santillan) [inline docs/semgrep-code/semgrep-assistant-code.md:57]:
```suggestion
* Semgrep Assistant logs and stores the GPT prompts and responses for the sake of performance evaluation, which include source code snippets.
```

File: docs/semgrep-code/semgrep-assistant-code.md
Comment (s-santillan) [inline docs/semgrep-code/semgrep-assistant-code.md:65]:
```suggestion
* When using Semgrep Assistant, source code **does** leave your repository; Assistant submits part of the file with a finding to OpenAI for processing by a GPT model. OpenAI is not allowed to use the submitted code for training their models.
```

**Review by s-santillan — APPROVED** [review]
LGTM! Let's just wait for Jack's approval too before merging.

**Review by jsmoxon — APPROVED** [review]
This looks good to me! Thank you so much!

**Pull request merged**
